RimWorld/RimWorld/NPC.cpp

```diff
@@ -0,0 +1,456 @@
+                    std::cout << currentTile.x << " " << currentTile.y << std::endl;
+                    std::cout << m_goalTile.x << " " << m_goalTile.y << std::endl;
+                    harvestResource();
+                }
+                else if (m_currentWorkType == WorkType::RESOURCE_TRANSFER)
+                {
+                    if (currentTile == m_goalNextTile)
+                    {
+                        transferResource();
+                    }
+                    else
+                    {
+                        std::vector<Resource>& resources = ResourceManager::GetInstance().m_newResources;
+                        for (auto& resource : resources)
+                        {
+                            if (resource.getPosition2u() == m_goalTile)
+                            {
+                                resource.setVisible(false);
+                                m_goalTile = m_goalNextTile;
+                                resource.setPosition(m_goalNextTile);
+                            }
+                        }
+                    }
+                }
+            }
+            else
+            {
+                m_goalTile = findRandomTile();
+                m_isWaiting = true;
+                m_pauseStartTime = std::chrono::steady_clock::now();
+                std::cout << "NPC reached the goal. Waiting for 5 seconds..." << std::endl;
+            }
+        }
+        else
+        {
+            updatePath(m_mapGraph);
+            const sf::Vector2u nextTileToMove = getNextTileToMove();
+            const sf::Vector2f targetPosition = Map::GetInstance().getPositionFromTile(nextTileToMove);
+            moveTo(dt, targetPosition, currentPosition);
+        }
+    }
+}
+
+void NPC::loadTexture()
+{
+    m_texture->loadFromFile("Assets/NPC.png");
+}
+
+void NPC::harvestResource() 
+{
+    if (!m_taskAssigned) return;
+
+    sf::Vector2u currentTile = Map::GetInstance().getTileFromPosition(m_sprite.getPosition());
+    std::vector<sf::Vector2u> nearbyTiles = getAdjacentTilesforMine(currentTile);
+
+    for (const auto& tile : nearbyTiles)
+    {
+        Resource* resource = ResourceManager::GetInstance().getResourceAt(tile);
+
+        if (resource && !resource->isDestroyed()) {
+            resource->destroy();
+            std::vector<std::pair<int, int>> resourceTiles = resource->getTiles();
+
+            for (const auto& resTile : resourceTiles) {
+                Map::GetInstance().setObjectLayer(resTile.first, resTile.second, 0);
+            }
+
+            int quantity = 0;
+            switch (resource->getType()) {
+            case ResourceType::TREE:
+                quantity = 15; 
+                break;
+            case ResourceType::STONE:
+                quantity = 5; 
+                break;
+            case ResourceType::METAL:
+                quantity = 3; 
+                break;
+            default:
+                quantity = 0;
+            }
```

Comment:
std::map<ResourceType, int> m_ResourceTypeToQuantityMap;

Reply:
setup it in c-tor, or read from file and just use that in form quantity = m_ResourceToQuantityMap[Resource];